docs/graphs/graph-concepts/depth-first-breadth-first.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Depth-first and Breadth-first Traversals
+sidebar_position: 80
+---
+
+Depth-first traversal (DFT) and breadth-first traversal (BFT) are two common strategies for exploring and navigating a graph, both used to visit all the vertices and edges of a graph in a systematic manner. Each traversal method has its specific use cases and characteristics.
+
+In the context of Macrometa's global data network, DFT and BFT can be used to explore relationships between vertices and edges within the graph, enabling complex graph analysis and pattern matching. To perform DFT and BFT on Macrometa graphs, you can use [C8QL](../../queries/c8ql/), Macrometa's query language.
+
+It is important to remember that both DFT and BFT can be resource-intensive operations, especially when traversing large graphs or exploring many levels deep. It is advisable to limit the traversal depth or use filters to reduce the number of vertices and edges visited during the traversal.
+
+For more information about querying graphs, refer to [Graph Queries](../graph-queries/).
+
+To learn more about querying in general, refer to [Queries](../../queries/).
+
+## Depth-First Traversal (DFT)
+
+Depth-first traversal starts at a given vertex and explores as far as possible along each branch before backtracking. In other words, DFT goes deep into the graph along a single path, visiting all the descendants of a vertex before returning to explore other branches. DFT can be implemented using recursion or an explicit stack data structure.
+
+### Steps for DFT
+
+1. Start at a source vertex.
+1. Visit the vertex and mark it as visited.
+1. For each adjacent unvisited vertex, perform a depth-first traversal from that vertex.
+1. Backtrack when there are no more unvisited adjacent vertices.
+
+DFT is particularly useful for tasks like searching for cycles in a graph or determining whether a path exists between two vertices.
+
+### DFT Example
+
+Suppose you have a graph named `socialGraph` with a vertex collection `users` and an edge collection `connections` representing the relationships between users. To perform a depth-first traversal starting from a user with the ID `users/12345` and explore up to a depth of 3, you can use the following C8QL query:
```

Comment:
Also here it would be good if we can give a proper example.. maybe point to some preloaded example graphs we have like "traversalGraph" instead of "socialGraph", also it would be "circles" collection here instead of "users" and instead of "connections" it would be "circles" .. we can use "circles/A" as the starting vertex..
and the subsequent query will be:

`FOR vertex, edge, path IN 1..3 ANY 'circles/A' GRAPH 'traversalGraph'
  RETURN {
    user: vertex._key,
    connection: edge ? edge._key : null,
    depth: LENGTH(path.edges)
  }`

Reply:
Basically just a suggestion to specify whatever sample data is to be used by the user to make the query running here , currently we don't specify how and what data to add in the respective collections

Reply:
Will do!